test/libsolidity/semanticTests/inlineAssembly/calldata_struct_assign_and_return.sol

```diff
@@ -0,0 +1,21 @@
+contract C{
+  struct S{ int8 x; int8 y; }
+  function f() internal pure returns(S calldata s) {
+    assembly {
+      s := 0x24
+    }
+  }
+  function g() public pure returns(int8, int8) {
+    S calldata s = f();
+    return (s.x, s.y);
+  }
+  function h() public pure returns(uint256) { f(); return 0x42; }
+  function i() public pure returns(uint256) { abi.decode(msg.data[4:], (S)); return 0x42; }
+}
+// ====
+// compileViaYul: also
+// ----
+// g(): 0xCAFFEE, 0x42, 0x21 -> 0x42, 0x21
+// g(): 0xCAFFEE, 0x4242, 0x2121 -> FAILURE
+// h() -> 0x42
+// i() -> FAILURE
```

Comment:
Yeah... these two are basically the same, but ``h()`` skips validation of the outermost layer in calldata, so it succeeds even with empty calldata, while ``i()`` performs this validation and reverts...

However, the same thing will apply for cases like the ``calldata_assign_form_nowhere`` test (https://github.com/ethereum/solidity/blob/dea1b9ec7995ff7f16cdee7925a5858c5ce33982/test/libsolidity/semanticTests/inlineAssembly/calldata_assign_from_nowhere.sol) and we do not appear to care for that case either...

Reply:
Or to be more precise: https://github.com/ethereum/solidity/blob/dea1b9ec7995ff7f16cdee7925a5858c5ce33982/test/libsolidity/semanticTests/inlineAssembly/calldata_assign_from_nowhere.sol will also succeed, if it sets 
``x.length := 64``... i.e. it will happily read beyond ``calldatasize()``...